include/zephyr/bluetooth/gap.h

```diff
@@ -825,13 +825,83 @@ enum {
 /** Maximum Periodic Advertising Interval (N * 1.25 ms) */
 #define BT_GAP_PER_ADV_MAX_INTERVAL             0xFFFF /* 81.91875 s */
 
+/** @brief Convert periodic advertising interval (N * 0.625 ms) to milliseconds */
+#define BT_GAP_ADV_INTERVAL_TO_MS(_int) ((_int) * 5U / 8U)
+
+/** @brief Convert isochronous interval (N * 1.25 ms) to milliseconds */
+#define BT_GAP_ISO_INTERVAL_TO_MS(_int) ((_int) * 5U / 4U)
+
```

Comment:
These also have lossy conversion, rounding down. We should have the note here as well. Maybe we should note that `BT_GAP_ADV_INTERVAL_TO_MS(1)` will round down to zero? I think rounding to zero is especially important to note because it's a divide-by-zero hazard.

We could avoid the lossy conversion by using us instead of ms. Of course we would have to change everything to be consistent. What do you think?

Reply:
> We could avoid the lossy conversion by using us instead of ms. Of course we might have to change everything to be consistent, and it would still be lossy the other way. What do you think?

Our API uses the various units (1.25ms or 0.625ms), which will always have the risk of being lossy when converting _to_ the units (e.g. 620us cannot be represented). 

However for the macros converting from units to ms, we can definitely consider microseconds instead, so that we properly represent the result. 
The reason why I went with MS here is that we have the `BT_GAP_PER_ADV_INTERVAL_TO_MS` and `BT_CONN_INTERVAL_TO_MS` macros already, but we do also have `BT_CONN_INTERVAL_TO_US`. 

Should I modify the added macros here to microseconds? 
Should we have both microseconds and millisecond versions of them like `BT_CONN_INTERVAL_TO_`?

Reply:
Added notes, but please let me know if I should modify it further to use microseconds instead

Reply:
> Maybe we should note that BT_GAP_ADV_INTERVAL_TO_MS(1) will round down to zero? I think rounding to zero is especially important to note because it's a divide-by-zero hazard.

Hmm, we really shouldn't have anything in our code that divides by 0 without checking first, and controller should reject as well. 

I'm a bit on the fence w.r.t. adding `BT_GAP_ADV_INTERVAL_TO_MS(1)` in the note, as an advertising interval of `1`ms is invalid (at least when using HCI).

Reply:
> Hmm, we really shouldn't have anything in our code that divides by 0 without checking first, and controller should reject as well.
> 
> I'm a bit on the fence w.r.t. adding `BT_GAP_ADV_INTERVAL_TO_MS(1)` in the note, as an advertising interval of `1`ms is invalid (at least when using HCI).

I wasn't worrying about the stack. I was worrying about any potential divide-by-zero in application code.

But since `1` is an impossible advertising interval, it does not really matter (unless it's not impossible, but just reserved for future).

Can we add the allowed input ranges to the documentation?

Reply:
I think microseconds is a good idea. @jhedberg, what do you think?

Reply:
> Can we add the allowed input ranges to the documentation?

Sure we can document those - But given that they are macros we can of course not return any error codes

Reply:
> I think microseconds is a good idea. @jhedberg, what do you think?

Sure, that works for me. Could be alongside of the _MS variants too.

Reply:
I'll add US versions; then we can decide whether to keep or toss the MS versions. The MS versions will just be XXX_TO_US / 1000